src/users_router.rs

```diff
@@ -64,3 +116,54 @@ impl UsersRouter {
         chain
     }
 }
+
+#[test]
+fn test_cors_allowed_endpoints() {
+    use std::net::ToSocketAddrs;
+    use self::iron::{method, TypeMap, Url};
+
+    // Stub request
+    fn request<'a, 'b>(method: &Method, path: &str) -> Request<'a, 'b> {
+        let path = "http://localhost:3000/".to_string() + path;
+        Request {
+            url: Url::parse(&path).unwrap(),
+            remote_addr:
+                "localhost:3000".to_socket_addrs().unwrap().next().unwrap(),
+            local_addr:
+                "localhost:3000".to_socket_addrs().unwrap().next().unwrap(),
+            headers: headers::Headers::new(),
+            body: unsafe { ::std::mem::uninitialized() },
+            method: method.clone(),
+            extensions: TypeMap::new()
+        }
+    }
+
+    // Test that all CORS allowed endpoints get the appropriate CORS headers.
+    for endpoint in CORS::ENDPOINTS {
+        let (ref method, path) = *endpoint;
+        let path = path.join("/");
```

Comment:
This becomes e.g. `permissions/_/*` and not something like `permissions/_/foo`, which is probably the reason this test is passing :)

You could replace all occurrences of the string `*` with the string `foo` to get a more realistic test.